### PR TITLE
Feature/justiceotuya/finishing time line

### PR DESCRIPTION
**Pull request by justiceotuya**

### CODE REVIEW

File: components/timeLine/utils.js [hunk 1]
```diff
@@ -1,7 +1,17 @@
 
-const PROFILE_URL = '../../static/data/timelineData.json';
+const TIMELINE_DATA_URL = '../../static/data/timelineData.json';
+const PROFILE_URL = '../../../static/data/profiles.json';
 
 export const fetchTimeLineData = async () => {
+    const response = await fetch(TIMELINE_DATA_URL);
+    const data = await response.json();
+    if (response.stats >= 400) {
```
Comment (EngrEric) [inline components/timeLine/utils.js]: what is `response.stats` ?

File: components/timeLine/actions.js [hunk 2]
```diff
@@ -25,6 +28,20 @@ export const setTimeLineData = payload => ({
     type: FETCH_TIMELINE_DATA_SUCCESS,
 });
 
+export const loadOnlineFriendsData = () => ({
+    type: FETCH_ONLINE_USERS_REQUEST,
+});
```
Comment (EngrEric) [inline components/timeLine/actions.js:33]: always make sure your action and its type potray similar meaning while naming them.
here you can say, `REQUEST_LOAD_ONLINE_FRIENDS_DATA`

File: components/timeLine/actions.js [hunk 3]
```diff
@@ -25,6 +28,20 @@ export const setTimeLineData = payload => ({
...
+
+export const setOnlineFriendsError = payload => ({
+    payload,
+    type: FETCH_ONLINE_USERS_FAILURE,
+});
```
Comment (EngrEric) [inline components/timeLine/actions.js:38]: here too

File: components/timeLine/utils.js (hunk 1 above)
Comment (justiceotuya) [reply components/timeLine/utils.js]: mistake, supposed to be `response.status`

File: components/timeLine/utils.js [hunk 4]
```diff
@@ -1,10 +1,20 @@
 
-const PROFILE_URL = '../../static/data/timelineData.json';
+const TIMELINE_DATA_URL = '../../static/data/timelineData.json';
+const PROFILE_URL = '../../../static/data/profiles.json';
 
 export const fetchTimeLineData = async () => {
+    const response = await fetch(TIMELINE_DATA_URL);
+    const data = await response.json();
+    if (response.status >= 400) {
+        throw new Error(data.errors);
```
Comment (enyeInc) [inline components/timeLine/utils.js]: why are you throwing errors here? or at all, you do know what happens if you throw an error on the frontend? You should be consuming your errors and handling them accordingly

File: components/timeLine/utils.js [hunk 5]
```diff
@@ -1,10 +1,20 @@
 
-const PROFILE_URL = '../../static/data/timelineData.json';
+const TIMELINE_DATA_URL = '../../static/data/timelineData.json';
+const PROFILE_URL = '../../../static/data/profiles.json';
 
 export const fetchTimeLineData = async () => {
+    const response = await fetch(TIMELINE_DATA_URL);
```
Comment (enyeInc) [inline components/timeLine/utils.js]: This pattern is wrong, use sagas to make async request, not a utils function.

File: components/timeLine/utils.js [hunk 6]
```diff
@@ -1,10 +1,20 @@
...
+    const data = await response.json();
+    if (response.status >= 400) {
+        throw new Error(data.errors);
+    }
+    return data;
+};
+
+export const fetchProfileData = async () => {
```
Comment (enyeInc) [inline components/timeLine/utils.js]: same comment on this function

File: components/timeLine/utils.js (hunk 5 above)
Comment (justiceotuya) [reply components/timeLine/utils.js]: I am just defining the async request in a util folder but calling it in the saga file.

File: components/timeLine/utils.js (hunk 6 above)
Comment (justiceotuya) [reply components/timeLine/utils.js]: I am just defining the async request in a util folder but calling it in the saga file.

File: components/timeLine/sagas.js [hunk 7]
```diff
@@ -2,23 +2,39 @@ import {
...
 const {
-    FETCH_TIMELINE_REQUEST,
+    REQUEST_LOAD_TIMELINE_DATA,
+    REQUEST_LOAD_ONLINE_FRIENDS_DATA,
 } = actionTypes;
 
 function* handleTimeLineDataLoad() {
     try {
-        const data = yield call(fetchTimeLineData);
+        const response = yield call(fetch, TIMELINE_DATA_URL);
```
Comment (enyeInc) [inline components/timeLine/sagas.js]: so `fetch` doesn't explicitly throw an error and that is by design. so you have to check that the response is okay
```
if (response.ok) { ... } else { ... }
```

File: components/timeLine/sagas.js [hunk 8]
```diff
@@ -2,23 +2,39 @@ import {
...
     } catch (error) {
-        yield put(setTimeLineError(error.toString()));
+        yield put(setTimeLineError(error));
+    }
+}
+
+function* handleProfileDataLoad() {
+    try {
+        const response = yield call(fetch, PROFILE_URL);
+        const data = yield response.json();
```
Comment (enyeInc) [inline components/timeLine/sagas.js:29]: same as the comment above

File: components/timeLine/sagas.js [hunk 9]
```diff
@@ -2,23 +2,39 @@ import {
...
+    REQUEST_LOAD_TIMELINE_DATA,
+    REQUEST_LOAD_ONLINE_FRIENDS_DATA,
 } = actionTypes;
 
 function* handleTimeLineDataLoad() {
-    try {
-        const data = yield call(fetchTimeLineData);
+    const response = yield call(fetch, TIMELINE_DATA_URL);
+    const data = yield response.json();
+    if (response.ok) {
```
Comment (enyeInc) [inline components/timeLine/sagas.js:18]: You can't access the data is the response is not ok.

**Review by enyeInc — APPROVED** [review]
LGTM

**Pull request merged**